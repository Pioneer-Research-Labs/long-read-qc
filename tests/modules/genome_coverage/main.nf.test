nextflow_process {

    name "Test Process genome_coverage"
    script "modules/genome_coverage/main.nf"
    process "genome_coverage"

    test("Should generate genome coverage stats") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = Channel.fromList([ tuple( [id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/mapped_inserts.bam'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/mapped_inserts.bam.bai'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/mapped_insert_stats.tsv'),
                )])
                 """
            }
        }

        then {
            assert process.success
            assert snapshot(
            ("$outputDir/Gateway_EmptyBarcodedLib1_20241217/genome_coverage.tsv"),
            ("$outputDir/Gateway_EmptyBarcodedLib1_20241217/genome_cov_stats.tsv")
            ).match()
        }

    }

}
