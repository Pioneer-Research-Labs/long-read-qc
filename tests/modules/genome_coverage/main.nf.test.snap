{
    "Should generate genome coverage stats": {
        "content": [
            "genome_coverage.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
            "genome_cov_stats.tsv:md5,919d17e37c043287be33051aeff4fe5d"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-15T12:29:16.438821"
    }
}